### PR TITLE
Documented the protocol changes related to impersonation

### PR DESCRIPTION
**Pull request by dotStart**

### CODE REVIEW

File: bolt/bolt-protocol-message-specification-4.md [hunk 1]
```diff
@@ -1463,13 +1465,327 @@ Example:
...
+An **Explicit Transaction** (`BEGIN`+`RUN`) does not carry any data in the `extra` field.
+
+For **Auto-commit Transaction** (`RUN`) the `extra` field carries:
+
+- The `bookmarks` is a list of strings containing some kind of bookmark identification e.g `["neo4j-bookmark-transaction:1", "neo4j-bookmark-transaction:2"]`. Default: `[]`.
+- The `tx_timeout` is an integer in that specifies a transaction timeout in ms. Default: server-side configured timeout.
+- The `tx_metadata` is a dictionary that can contain some metadata information, mainly used for logging. Default: `null`.
+- The `mode` specifies what kind of server the `RUN` message is targeting. For write access use `"w"` and for read access use `"r"`. Default: `"w"`.
+- The `db` key specifies the database name for multi-database to select where the transaction takes place. `null` and `""` denote the server-side configured default database. Default: `null`.
+- The `imp_user` key specifies the impersonated user which executes this transaction. `null` denotes no impersonation (execution takes place as the current user). Default: `null`.
```
Comment (robsdedude) [inline bolt/bolt-protocol-message-specification-4.md:1532]: Maybe add a new or alter an existing example to include `db` and `imp_user`.

File: bolt/bolt-protocol-message-specification-4.md [hunk 2]
```diff
@@ -1463,13 +1465,327 @@ Example:
...
+  imp_user::String
+)
+```
+
+- The `bookmarks` is a list of strings containing some kind of bookmark identification e.g `["neo4j-bookmark-transaction:1", "neo4j-bookmark-transaction:2"]`. Default: `[]`.
+- The `tx_timeout` is an integer in that specifies a transaction timeout in ms. Default: server-side configured timeout.
+- The `tx_metadata` is a dictionary that can contain some metadata information, mainly used for logging. Default: `null`.
+- The `mode` specifies what kind of server the `RUN` message is targeting. For write access use `"w"` and for read access use `"r"`. Defaults to write access if no mode is sent. Default: `"w"`.
+- The `db` key specifies the database name for multi-database to select where the transaction takes place.  `null` and `""` denote the server-side configured default database. Default: `null`.
+- The `imp_user` key specifies the impersonated user which executes this transaction. `null` denotes no impersonation (execution takes place as the current user). Default: `null`.
```
Comment (robsdedude) [inline bolt/bolt-protocol-message-specification-4.md:1638]: Maybe an example using `imp_user` here as well?

File: bolt/bolt-protocol-message-specification-4.md [hunk 3]
```diff
@@ -1463,13 +1465,327 @@ Example:
...
+
+#### Server Response `SUCCESS`
+
+A `SUCCESS` message response indicates that the client is permitted to exchange further messages.
+
+The following fields are defined for inclusion in the `SUCCESS` metadata.
+
+- `rt::Dictionary(ttl::Integer, db::String, servers::List<Dictionary(addresses::List<String>, role::String)>)`, the current routing table.
+    - `ttl::Integer` specifies for how many seconds this routing table should be considered valid.
+    - `db::String` identifies the database for which this routing table applies.
```
Comment (robsdedude) [inline bolt/bolt-protocol-message-specification-4.md:1753]:
```suggestion
    - `db::String` identifies the database to which this routing table applies.
```
not 100% sure

Comment (dotStart) [reply bolt/bolt-protocol-message-specification-4.md:1753]: Both are fine in this context.

**Pull request merged**